exe/shell/main.go

```diff
@@ -218,88 +269,123 @@ func addressCmd() (out string, err error) {
 	return
 }
 
-func threadCmd(args []string) (out string, err error) {
-	if len(args) == 0 {
-		err = fmt.Errorf("enter a thread name to use")
+func threadsCmd() (out string, err error) {
+	q, err := ds.Query(query.Query{Prefix: "/names"})
+	if err != nil {
+		return
+	}
+	all, err := q.Rest()
+	if err != nil {
 		return
 	}
-	name := args[0]
 
-	var addr ma.Multiaddr
-	if len(args) > 1 {
-		addr, err = ma.NewMultiaddr(args[1])
+	var id thread.ID
+	for i, e := range all {
+		id, err = thread.Cast(e.Value)
 		if err != nil {
 			return
 		}
-		if !canDial(addr) {
-			return "", fmt.Errorf("address is not dialable")
+		name := e.Key[strings.LastIndex(e.Key, "/")+1:]
+		out += blue(name) + grey(" ("+id.String()+")")
+		if i != len(all)-1 {
+			out += "\n"
 		}
 	}
+	return
+}
 
-	x, err := ds.Get(datastore.NewKey("/names/" + name))
-	if err == datastore.ErrNotFound {
-		if addr != nil {
-			info, err := api.AddThread(ctx, addr)
-			if err != nil {
-				return "", err
-			}
-			threadID = info.ID
-		} else {
-			threadID = thread.NewIDV1(thread.Raw, 32)
-		}
-		err = ds.Put(datastore.NewKey("/names/"+name), threadID.Bytes())
+func enterCmd(name string) (out string, err error) {
+	idv, err := ds.Get(datastore.NewKey("/names/" + name))
```

Comment:
Since you use this prefix a bunch here, you could probably just create a const to make it easier to reference?